km/km_vcpu_run.c

```diff
@@ -515,7 +517,7 @@ static void km_vcpu_one_kvm_run(km_vcpu_t* vcpu)
               vcpu->regs.rsp,
               vcpu->sregs.cr2);
       vcpu->cpu_run->exit_reason = KVM_EXIT_INTR;
-      switch (errno) {
+      switch (ioctl_errno) {
```

Comment:
I think you should restore errno.
The invocation of km_err() in the default cause wants to use errno.